Sources/Danger/Danger.swift

```diff
@@ -74,25 +74,45 @@ public func warn(_ message: String) {
     DangerRunner.shared.results.warnings.append(Violation(message: message))
 }
 
+/// Adds an inline warning message to the Danger report
+public func warn(message: String, file: String, line: Int) {
+    DangerRunner.shared.results.warnings.append(Violation(message: message, file: file, line: line))
+}
+
 /// Adds a warning message to the Danger report
 ///
 /// - Parameter message: A markdown-ish
 public func fail(_ message: String) {
     DangerRunner.shared.results.fails.append(Violation(message: message))
 }
 
+/// Adds an inline fail message to the Danger report
+public func fail(message: String, file: String, line: Int) {
```

Comment:
Any reason you chose to use function overloading instead of just making file & string parameters optional with defaults?

Reply:
Oh, yeah. I just don't think we should let users add the file without a line. Although it's possible to do (in danger-js, currently, because I couldn't find a better way without over-engineering), it doesn't do much (will just end up in the main comment anyways).

Reply:
Great point 👍 Was just curious and wanted to understand your reasoning

Reply:
Totally, didn't think about that before, good question 👍

Reply:
Yeah, I like this 👍